src/components/links/LinkBase.tsx

```diff
@@ -30,14 +30,7 @@ import {
 import { isSonarLink } from '~common/helpers/url';
 import { IconLinkExternal } from '../icons/IconLinkExternal';
 
-type RouterNavLinkPropsAllowed =
-  | 'download'
-  | 'end'
-  | 'reloadDocument'
-  | 'state'
-  | 'style'
-  | 'title'
-  | 'to';
+type RouterNavLinkPropsAllowed = 'download' | 'reloadDocument' | 'state' | 'style' | 'title' | 'to';
```

Comment:
@jeremy-davis-sonarsource this would be a "breaking change" in that users would now have to pass `matchesExactPath` instead of `end`, and the logic is reversed (defaults to `true) - WDYT?

Reply:
It should default to false and be called something like `isMatchingExactPath`:
https://docs.google.com/document/d/1r50MkSdJEznh8ijjpgRoJffhQ9IKj5tM41Y2XbwEmO0/edit#heading=h.n71kn868u7ey

Reply:
OK, I've renamed and defaulted to `false` in `LinkBase`. I've defaulted to `true` in `DropdownMenuItemLink`, however, because it looks like it's the more common case. WDYT?

Reply:
Then I'd call it `isMatchingPartialPath` and make it default to false.

Reply:
reverting this to default to a partial match like react-router, as discussed